### PR TITLE
build: use +~ for cross-building

### PR DESCRIPTION
**Pull request by jrudolph**
Porting raboof's command from https://github.com/akka/akka/pull/30981/files

### CODE REVIEW

**Comment by jrudolph** [comment]
Scala 3 scripted test fails because the `_~` cannot parse with arguments.

**Comment by raboof** [comment]
> Scala 3 scripted test fails because the `+~` cannot parse with arguments.

Hmm, I hoped at least `sbt "+~ 2.13 \"sbt-akka-grpc/scripted scala3/*\""` would do the trick (possibly by adding extra `"` to the `++` invocation inside), but it looks like that doesn't work either..

File: project/Common.scala [hunk 1]
```diff
@@ -77,4 +78,30 @@ object Common extends AutoPlugin {
...
+        parsed.resultEmpty match {
+          case e: sbt.internal.util.complete.Parser.Failure =>
+            throw new IllegalStateException(e.errors.mkString(", "))
+          case sbt.internal.util.complete.Parser.Value(v) =>
+            v()
+        }
+      }
+      val commands = s"++$requestedVersion" +: args.tail
+      commands.foldLeft(initialState)(run)
+    }
```
Comment (He-Pin) [inline project/Common.scala:105]: it would be great to port this to sbt too.

**Review by raboof — APPROVED** [review]

File: project/Common.scala (hunk 1 above)
Comment (raboof) [reply project/Common.scala:105]: I would like that, though switching has been a much-discussed topic over the years - I'm curious if people would like this approach. I opened the discussion at https://github.com/sbt/sbt/discussions/6893

Comment (jrudolph) [reply project/Common.scala:105]: Alternatively, we could consider moving to https://github.com/sbt/sbt-projectmatrix in the future.

Comment (raboof) [reply project/Common.scala:105]: > Alternatively, we could consider moving to https://github.com/sbt/sbt-projectmatrix in the future.

Yeah, that might be interesting in any case

**Pull request merged**